shared/src/main/java/eu/darkbot/shared/modules/LootModule.java

```diff
@@ -0,0 +1,271 @@
+            distance = (radius += radiusFix);
+            // Moved distance + speed - distance to chosen radius same angle, divided by radius
+            angleDiff = Math.max(/*(hero.shipInfo.speed * 0.625) +*/ (min(200, target.getLocationInfo().getSpeed()) * 0.625)
+                    - hero.distanceTo(Location.of(targetLoc, angle, radius)), 0) / radius;
+        }
+        direction = getBestDir(targetLoc, angle, angleDiff, distance);
+
+        while (!movement.canMove(direction) && distance < 10000)
+            direction.toAngle(targetLoc, angle += backwards ? -0.3 : 0.3, distance += 2);
+        if (distance >= 10000) direction.toAngle(targetLoc, angle, 500);
+
+        setConfig(direction);
+
+        movement.moveTo(direction);
+    }
+
+    protected Location getBestDir(Locatable targetLoc, double angle, double angleDiff, double distance) {
+        int maxCircleIterations = this.maxCircleIterations.getValue(), iteration = 1;
+        double forwardScore = 0, backScore = 0;
+        do {
+            forwardScore += score(Locatable.of(targetLoc, angle + (angleDiff * iteration), distance));
+            backScore += score(Locatable.of(targetLoc, angle - (angleDiff * iteration), distance));
+            // Toggle direction if either one of the directions is perfect, or one is 300 better.
+            if (forwardScore < 0 != backScore < 0 || Math.abs(forwardScore - backScore) > 300) break;
+        } while (iteration++ < maxCircleIterations);
+
+        if (iteration <= maxCircleIterations) backwards = backScore > forwardScore;
+        return Location.of(targetLoc, angle + angleDiff * (backwards ? -1 : 1), distance);
+    }
+
+    protected double score(Locatable loc) {
+        return (movement.canMove(loc) ? 0 : -1000) - npcs.stream() // Consider barrier as bad as 1000 radius units.
+                .filter(n -> target != n)
+                .mapToDouble(n -> Math.max(0, n.getInfo().getRadius() - n.distanceTo(loc)))
+                .sum();
+    }
+
+    protected void setConfig(Locatable direction) {
+        if (!attack.hasTarget()) hero.setRoamMode();
+        else if (runConfigInCircle.getValue()
+                && target.getHealth().hpPercent() < 0.25
+                && hero.distanceTo(direction) > target.getInfo().getRadius() * 2) hero.setRunMode();
+        else if (hero.distanceTo(direction) > target.getInfo().getRadius() * 3) hero.setRoamMode();
+        else hero.setAttackMode(target);
+    }
+
+    protected boolean isAttackedByOthers(Npc npc) {
+        for (Ship ship : this.ships) {
+            if (!ship.isAttacking(npc)) continue;
+
+            if (!npc.getInfo().hasExtraFlag(NpcExtraFlag.IGNORE_ATTACKED)) npc.setBlacklisted(20_000);
+            return true;
+        }
+        return false;
+    }
+
+    protected Npc closestNpc(Locatable location) {
+        //cant use attack.hasTarget() or isLocked() here cause can return true even if getTargetAs returns null
```

Comment:
use target as a local variable in this method instead of relying on it having been set prior to calling